File: Leetcode July Challenge/Python/Reduce Array Size to The Half.py
from collections import defaultdict
class Solution:
    def minSetSize(self, arr: List[int]) -> int:
        
        d = defaultdict(int)
        for i in range(len(arr)):
            d[arr[i]] += 1
            
        
        l = list(d.values())
        l.sort(reverse = True)
        
        count = 0
        ans = 0
        index = 0
        size =  len(arr)
        while count*2 < size:
            ans += 1
            count += l[index]
            index += 1
        
        return ans
        
        
        

File: Leetcode July Challenge/C++/Reduce Array Size to The Half.cpp
#include<bits/stdc++.h>
class Solution {
public:
   
    
    int minSetSize(vector<int>& arr) {
        unordered_map<int,int> m;
        for(int i = 0; i< arr.size(); i++){
            m[arr[i]] += 1;
        }
        
        multiset<int, greater<int>> s;
        
        for(auto n : m){
            s.insert(n.second);
        }
        
        int size = arr.size();
        int count = 0;
        int ans = 0;
        
        for(auto it = s.begin(); count * 2 < size ; it++){
            ans++;
            count += *it;
        }
        
        return ans;
        
    }
};
